Library/brew.rb

```diff
@@ -97,7 +97,7 @@ def require?(path)
 
   cmd = HOMEBREW_INTERNAL_COMMAND_ALIASES.fetch(cmd, cmd)
 
-  sudo_check = %w[ install link pin unpin upgrade ]
+  sudo_check = %w[ install reinstall postinstall link pin unpin upgrade ]
```

Comment:
`pin` is already in there. Agree about the rest; all would/could create or symlink files/folders that would then fail to remove without another application of `sudo`.